packages/data/src/redux-store/test/index.js

```diff
@@ -235,3 +235,45 @@ describe( 'controls', () => {
 		} );
 	} );
 } );
+
+describe( 'resolveSelect', () => {
+	let registry;
+	let shouldFail;
+
+	beforeEach( () => {
+		shouldFail = false;
+		registry = createRegistry();
+
+		registry.registerStore( 'store', {
+			reducer: ( state = null ) => {
+				return state;
+			},
+			selectors: {
+				getItems: () => 'items',
+			},
+			resolvers: {
+				getItems: () => {
+					if ( shouldFail ) {
+						throw new Error( 'cannot fetch items' );
+					}
+				},
+			},
+		} );
+	} );
+
+	it( 'resolves when the resolution succeeded', async () => {
+		shouldFail = false;
+		const promise = registry.resolveSelect( 'store' ).getItems();
+		jest.advanceTimersByTime( 1 );
```

Comment:
This is oddly specific, maybe we could do `jest.runAllTimers()` or `jest.runAllTicks()`?